File: by-kotlin/kt-boj/solutions/4375-1.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.math.BigInteger

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    while (true) {
        val input = readLine() ?: break

        val k = input.toInt().toBigInteger()

        var num = BigInteger.ONE
        var cnt = 1

        while (true) {
            if (num.mod(k) == BigInteger.ZERO) break

            num = num.multiply(BigInteger("10")).plus(BigInteger.ONE)
            cnt += 1
        }

        println(cnt)
    }

}

File: by-python/boj/4375-1.py
def solution(n):
    cnt = 1
    num = 1
    while True:
        if num % n == 0:
            return cnt
        num = num * 10 + 1
        cnt += 1


if __name__ == '__main__':
    while True:
        try:
            n = int(input())
        except EOFError:
            exit(0)

        print(solution(n))
